app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -43,22 +87,32 @@ public function createOrder($order)
 
         $currency = Razorpay_Payments_Model_Paymentmethod::CURRENCY;
 
-        $url = $this->getRelativeUrl('order');
+        $razorpayOrderId = Mage::getSingleton('core/session')->getRazorpayOrderID();
 
-        $data = array(
-            'receipt'         => $orderId,
-            'amount'          => $amount,
-            'currency'        => $currency,
-            'payment_capture' => 1
-        );
+        if ($razorpayOrderId === null)
+        {
+            Mage::log(['razorpayOrderId' => 'NULL']);
+        }
+
+        if (($razorpayOrderId === null) or 
+            (($razorpayOrderId) and ($this->verifyOrderAmount($razorpayOrderId, $order) === false)))
+        {
+            $data = $this->getRazorpayOrderData($order);
+
+            $data['payment_capture'] = 1;
```

Comment:
`getRazorpayOrderData` is doing 2 different things:

- get request params for order creation
- return expected order data

(2) doesn't necessarily have to be the same as (1), which makes it confusing. Improve by using 2 functions instead that wrap?